src/AmbientSounds.Uwp/Views/ShellPage.xaml.cs

```diff
@@ -38,6 +38,27 @@ public ShellPage()
         // Set custom title bar.
         // Source: https://learn.microsoft.com/en-us/windows/apps/develop/title-bar?tabs=winui2#interactive-content 
         Window.Current.SetTitleBar(AppTitleBar);
+
+        Windows.UI.Core.Preview.SystemNavigationManagerPreview.GetForCurrentView().CloseRequested +=
+        async (sender, args) =>
+        {
+            args.Handled = true;
+
+            bool playing = App.Services.GetService<IMixMediaPlayerService>().PlaybackState == MediaPlaybackState.Playing ? true : false;
+
+            if (playing)
+            {
+                bool result = await ViewModel.OpenConfirmCloseAsync();
+                if (result)
+                {
+                    App.Current.Exit();
+                }
+            }
+            else
+            {
+                App.Current.Exit();
```

Comment:
Maybe use Application.Current instead?

Reply:
What's the difference?

Reply:
VS warns for it iirc, also it isn't application specific functionality that's being overridden here.

Reply:
VS doesn't throw a warning for `App.Current.Exit()`. I think it's fine to leave it as-is.

Reply:
Actually yeah, it throws an IDE warning. I'll swap to Application. Thanks!